packages/funding-rate-proposer/src/proposer.js

```diff
@@ -278,12 +275,104 @@ class FundingRateProposer {
         at: "PerpetualProposer#updateFundingRate",
         message: "Skipping proposal because current rate is within allowed margin of error",
         fundingRateIdentifier,
-        proposedRate: offchainFundingRate,
+        proposedRate: pricefeedPrice,
         currentRate: onchainFundingRate,
         allowedError: this.fundingRateErrorPercent
       });
     }
   }
+
+  // If there is a pending funding rate that can be published, AND the current offchain price differs from the pending
+  // rate more than some threshold, than publish the pending rate
+  async _applyFundingRate(contractAddress) {
+    const cachedContract = this.contractCache[contractAddress];
+    const currentFundingRateData = cachedContract.state.currentFundingRateData;
+    const fundingRateIdentifier = this.hexToUtf8(currentFundingRateData.identifier);
+
+    // Get `getSettleableProposals` in which the proposer was this bot account. Return early if there are no
+    // pending proposals.
+    const publishableProposals = this.optimisticOracleClient.getSettleableProposals(this.account).filter(proposal => {
```

Comment:
it only considers rates that are proposed by _this_ bot? can you comment on this?

Reply:
I can generalize this actually, it is currently set like this because `getSettleableProposals` expects an `account` to be passed. I'll change the OO client to allow the caller to get proposals for ALL accounts

Reply:
I see. that's probs fine. it's just that this might need to be called on proposals that were made by another bot, right? ideally it should be generalised but this can probs be put in the backlog for later.

Reply:
This is no longer relevant as I've refactored this PR to simulate the `applyFundingRate` method